lib/ex_doc/language/erlang.ex

```diff
@@ -572,6 +572,10 @@ defmodule ExDoc.Language.Erlang do
             {{:., _, [module, name]}, _, args}, acc ->
               {{:t, [], args}, [{pp({module, name}), {module, name, length(args)}} | acc]}
 
+            ## type module.type/0
+            {:., _, [module, name]} = ast, acc ->
+              {ast, [{pp({module, name}), {module, name, 0}} | acc]}
```

Comment:
I'm going to look at this again because we make an assumption here, which may be fine, but we're assuming that the next element in the tuple is going to be an empty list, which isn't great. We should perhaps catch all this at field type or possibly do a look behind and look ahead.